packages/frontend/types/api/postSelect.ts

```diff
@@ -0,0 +1,58 @@
+// Select query for the route GET /posts/{postId}
+
+import { Prisma } from '@prisma/client';
+
+const selectFields = {
+  id: true,
+  title: true,
+  body: true,
+  timestamp: true,
+  userId: true,
+  upvotes: {
+    select: {
+      id: true,
+      address: true,
+      timestamp: true,
+    },
+  },
+};
+
+// We need to hardcode the structure of the query to let Prisma infer the return types.
+// (i.e. we can't use recursive function calls to construct this object)
```

Comment:
I was afraid this might be the case. Might be a situation where we're ok trading off type safety for better recursion.